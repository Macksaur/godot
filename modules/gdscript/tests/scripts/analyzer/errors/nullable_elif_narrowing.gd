func test():
    var nullable_string: String? = null
    if nullable_string == "2134":
        print(nullable_string.begins_with("asd")) # safe
    elif nullable_string == null:
        print(nullable_string.begins_with("asd")) # Not safe
    else:
        print(nullable_string.begins_with("asd")) # safe