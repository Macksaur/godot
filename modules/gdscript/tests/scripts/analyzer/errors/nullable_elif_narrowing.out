GDTEST_ANALYZER_ERROR
Potentially unsafe for nullable "nullable_string" to call "begins_with()".
